In [80]:
import pandas as pd
import numpy as np
import requests

In [81]:
df = pd.read_html('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M')[0]

In [82]:
df.head(12)

Postal Code           Borough                                 Neighborhood
0          M1A      Not assigned                                 Not assigned
1          M2A      Not assigned                                 Not assigned
2          M3A        North York                                    Parkwoods
3          M4A        North York                             Victoria Village
4          M5A  Downtown Toronto                    Regent Park, Harbourfront
5          M6A        North York             Lawrence Manor, Lawrence Heights
6          M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government
7          M8A      Not assigned                                 Not assigned
8          M9A         Etobicoke      Islington Avenue, Humber Valley Village
9          M1B       Scarborough                               Malvern, Rouge
10         M2B      Not assigned                                 Not assigned
11         M3B        North York                                    Don Mills

In [83]:
indexNames = df[df['Borough'] == 'Not assigned'].index

df1 = df.drop (indexNames, inplace=True)

In [84]:
indexNamesNeigh = df[df['Neighborhood'] == 'Not assigned'].index
indexNamesNeigh


Int64Index([], dtype='int64')

In [85]:
df.shape

(103, 3)

I read the table in wikipedia using pandas and put it into a frame
i removed the rows where the values of borough are not assigned
i checked if any neighborhood values has a not assigned value and still available
i used df.shape in order to know that we have 103 rows left.

In [63]:
!conda install -c conda-forge geocoder

Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - geocoder


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ratelim-0.1.6              |             py_2           6 KB  conda-forge
    ca-certificates-2020.4.5.1 |       hecc5488_0         146 KB  conda-forge
    certifi-2020.4.5.1         |   py36h9f0ad1d_0         151 KB  conda-forge
    geocoder-1.38.1            |             py_1          53 KB  conda-forge
    openssl-1.1.1g             |       h516909a_0         2.1 MB  conda-forge
    python_abi-3.6             |          1_cp36m           4 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         2.5 MB

The following NEW packages will be INSTALLED:

    geocoder:        1.38.1-py_1       conda-forge
    python_abi:    

In [86]:
dfll=pd.read_csv('http://cocl.us/Geospatial_data')

In [68]:
dfll.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [92]:
dfmerged = pd.merge(df, dfll, on="Postal Code")

In [93]:
dfmerged.shape

(103, 5)

In [95]:
dfmerged.head()

Postal Code           Borough                                 Neighborhood  \
0         M3A        North York                                    Parkwoods   
1         M4A        North York                             Victoria Village   
2         M5A  Downtown Toronto                    Regent Park, Harbourfront   
3         M6A        North York             Lawrence Manor, Lawrence Heights   
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government   

    Latitude  Longitude  
0  43.753259 -79.329656  
1  43.725882 -79.315572  
2  43.654260 -79.360636  
3  43.718518 -79.464763  
4  43.662301 -79.389494